File: diretorioprincipal/questao7.go
//atividade 06/09/24
//aluno: João Pedro Araújo Inácio
//questao7

package main

import "fmt"

type Professor struct {
	nome     string
	escolas  []*Escola
}

type Escola struct {
	nome       string
	professores []*Professor
}

func (p *Professor) AdicionarEscola(e *Escola) {

	for _, escola := range p.escolas {
		if escola == e {
			return
		}
	}
	p.escolas = append(p.escolas, e)
	e.AdicionarProfessor(p)
}

func (e *Escola) AdicionarProfessor(p *Professor) {

	for _, professor := range e.professores {
		if professor == p {
			return
		}
	}
	e.professores = append(e.professores, p)
}

func main() {

	escola1 := &Escola{nome: "motiva"}
	escola2 := &Escola{nome: "evolucao"}

	professor1 := &Professor{nome: "hailton"}
	professor2 := &Professor{nome: "job"}

	professor1.AdicionarEscola(escola1)
	professor1.AdicionarEscola(escola2)
	professor2.AdicionarEscola(escola1)

	fmt.Println("professores na", escola1.nome)
	for _, professor := range escola1.professores {
		fmt.Println(professor.nome)
	}

	fmt.Println("escolas do", professor1.nome)
	for _, escola := range professor1.escolas {
		fmt.Println(escola.nome)
	}
}

File: diretorioprincipal/questao7.py
#atividade 06/09/24
#aluno: João Pedro Araújo Inácio
#questao7

class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionar_escola(self, escola):
        if escola not in self.escolas:
            self.escolas.append(escola)
            escola.adicionar_professor(self)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        if professor not in self.professores:
            self.professores.append(professor)

escola1 = Escola("motiva")
escola2 = Escola("evolucao")

professor1 = Professor("Job")
professor2 = Professor("hailton")

professor1.adicionar_escola(escola1)
professor1.adicionar_escola(escola2)
professor2.adicionar_escola(escola1)

print(f"Professores na {escola1.nome}: {[prof.nome for prof in escola1.professores]}")
print(f"Professores na {escola2.nome}: {[prof.nome for prof in escola2.professores]}")
print(f"Escolas do {professor1.nome}: {[esc.nome for esc in professor1.escolas]}")